File: server/src/main/java/com/macina/room/entities/workoutDay/WorkoutDayDTO.java
package com.macina.room.entities.workoutDay;

import com.macina.room.crud.BaseDTO;
import com.macina.room.entities.workoutDayExercise.WorkoutDayExerciseDTO;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.List;

@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
public class WorkoutDayDTO extends BaseDTO {
    private short sequence;
    private Long workoutPlanId;
    private List<WorkoutDayExerciseDTO> exercises;
}

File: gymBro_client/src/app/models/workoutDay.dto.ts
import { BaseDTO } from './baseDto';
import { WorkoutDayExerciseDTO } from './workoutDayExercise.dto';

export interface WorkoutDayDTO extends BaseDTO {
  name: string;
  sequence: number;
  workoutPlanId: number;
  exercises: WorkoutDayExerciseDTO[];
}
